lib/shared-class.js

```diff
@@ -191,47 +191,34 @@ SharedClass.prototype.resolve = function(resolver) {
 };
 
 /**
- * Find a sharedMethod with the given name or function object.
+ * Find a sharedMethod with the given name or prototype name.
  *
- * @param {String|Function} fn The function or method name
- * @param {Boolean} [isStatic] Required if `fn` is a `String`.
- * Only find a static method with the given name.
+ * @param {String} methodName The method name or prototype method
```

Comment:
Interesting. I didn't realise `find` was supporting both name-based lookup and function-based lookup. I think we should preserve function-based lookup implementation somewhere, as we need it `SharedClass.prototype.methods`. I suppose it's ok to keep it in an private method only.

To simplify upgrade path, I would like to keep `SharedClass.prototype.find` with a deprecation warning (we use `depd` for that). Also to make the distinction between function-based and name-based lookup clear, I am proposing to rename `findMethod` to `findMethodByName`.

```js
var deprecated = require('depd')('strong-remoting'); // at the top

SharedClass.prototype.find = function(fn, isStatic) {
  deprecated('SharedClass.prototype.find is deprecated. Use SharedClass.prototype.findMethodByName instead.');
  // ...
};
```